solr/core/src/test/org/apache/solr/handler/admin/api/V2CollectionAPIMappingTest.java

```diff
@@ -87,13 +96,15 @@ public void testGetCollectionStatus() throws Exception {
 
 
   @Test
-  public void testRenameCollectionAllParams() throws Exception {
-    final SolrParams v1Params = captureConvertedV1Params(
-            "/collections/collName", "POST", "{\"rename\": {\"to\": \"targetColl\"}}");
+  public void testRenameCollectionAllParams(){
+    final SolrQueryRequest request = runRenameCollectionsApi("/collections/collName");
```

Comment:
`Test result returns  java.lang.AssertionError: expected:<rename> but was:<null>` 

What if we mock `SolrQueryRequest` and `SolrQueryResponse` and pass them to the`public void renameCollection(SolrQueryRequest req, SolrQueryResponse rsp)` method of the API then perform the assertions

Reply:
I will try to implement that, thanks

Reply:
> `Test result returns java.lang.AssertionError: expected:<rename> but was:<null>`
> 
> What if we mock `SolrQueryRequest` and `SolrQueryResponse` , pass them to the`public void renameCollection(SolrQueryRequest req, SolrQueryResponse rsp)` method of the API then perform the assertions

Please I need more clarification on this

Reply:
+1 to what Josh said.  AddReplicaPropertyAPITest is a good example on how to use mocks, but it's setup for APIs that use the JAX-RS framework, so you wouldn't want to follow it too closely.  But it's good if you're just looking for an example on the mock stuff.

And I'm sure we'd both be happy to clarify if you can elaborate a bit on **what** you need clarified.

[Q] One thing that jumps out at me on this line is the `runRenameCollectionsAPI`: is there a reason you've created a separate method for this instead of reusing one of the `captureConvertedV1Params` methods that's already out there.  If there's not some reason for `runRenameCollectionsAPI` that I'm missing, you could presumably just replace its usage here with a snippet like:

```
    final SolrParams v1Params = captureConvertedV1Params("/collections/collName/commands/rename", "POST",
            "{\"to\": \"targetColl\", \"async\": \"requestTrackingId\", \"followAliases\": true}");
```

The second thing that jumps out at me here is the path being sent as an argument to this method - it doesn't match the path of the API you created in this PR (see [here](https://github.com/apache/solr/pull/1080/files#diff-e1a70c63cd0e418b5027514b9833ab6ebef6d4dd71026a7e440d9c061ef181c6R60)).

Reply:
I thought you said we were moving away from the old captureConvertedV1Params syntax, I wasn't aware there are different method implementations

Reply:
Ah, not quite.  Or at least, I didn't mean to suggest that?

What I probably meant to say is that Solr is moving away from its traditional v2 API framework to a framework based on JAX-RS.  `captureConvertedV1Params` is a nice utility for testing APIs implemented using the "traditional" (i.e. non JAX-RS) framework.

In the long run I'd expect to see usage of `captureConvertedV1Params` decline, but not because it's deprecated or shouldn't be used - just because things will switch over to JAX-RS as time goes by.  This PR uses the traditional framework (which is absolutely still valid), so it's fine/normal to use `captureConvertedV1Params` in your tests.

I'll push up a commit here to help get the unit test over the line and then hopefully we can get this merged 👍

Reply:
It's clearer now, and I'm so glad to hear that this will be merged thank you.